packages/flutter_test/lib/src/widget_tester.dart

```diff
@@ -183,6 +197,24 @@ void testWidgets(
   }
 }
 
+WidgetTesterCallback _wrapWithLeakTracking(
+  String description,
+  WidgetTesterCallback callback,
+  LeakTesting? leakTesting,
+) {
+  // It is important to resolve settings outside of callback, to take
+  // value of LeakTesting.settings at the moment of this function execution,
+  // not at the moment of callback execution.
+  final LeakTesting settings = leakTesting ?? LeakTesting.settings;
+
+  Future<void> wrappedCallBack(WidgetTester tester) async {
+    mayBeSetupLeakTrackingForTest(settings, description);
+    await callback(tester);
+    ignoreAllLeaks();
```

Comment:
ignoreAllLeaks should be replaced with method like `finalizeTest`, defined in leak_tracker, that:

1. Declare all not disposed objects as leaks (it should happen for each test, not one time at the end, because we want to avoid flakiness) 
2. Pause leak tracking